examples/paper-invoice-to-inspectionlot/main.bal

```diff
@@ -0,0 +1,106 @@
+import ballerina/http;
+import ballerina/log;
+import ballerinax/sap.s4hana.api_inspectionlot_srv as inspectionLot;
+import ballerina/io;
+
+const OCR_REQUEST_PATH = "/invoice_parser";
+const OCR_URL = "https://api.edenai.run/v2/ocr";
+const CONTENT_TYPE = "Content-Type";
+const APPLICATION_JSON = "application/json";
+
+configurable S4HANAClientConfig s4hanaClientConfig = ?;
+configurable string ocrToken = ?;
+configurable string invoiceUrl = ?;
+
+final map<string> & readonly supplierMap = {"SNG Engineering Inc": "1010"};
+final map<string> & readonly plantMap = {"SNG Engineering Inc": "1010"};
+final map<string> & readonly materialMap = {"89FG": "FG376", "89QM": "E002"};
+
+final http:Client ocrHttpClient = check getOCRHttpClient();
+
+final inspectionLot:Client inspectionLotClient = check new (
+    config = {
+        auth: {
+            username: s4hanaClientConfig.username,
+            password: s4hanaClientConfig.password
+        }
+    },
+    hostname = s4hanaClientConfig.hostname
+);
+
+public function main() {
+
+    PaperInvoice|error invoiceResponse = readPaperInvoice();
+    if invoiceResponse is error {
+        log:printError("Error while reading paper invoice: " + invoiceResponse.message());
+        return;
+    }
+
+    log:printInfo("Received inspection lot with invoice number: " + invoiceResponse.invoice_number + " on " + invoiceResponse.date);
+    SAPInspectionLot|error inspectionLotData = transformInspectionLotData(invoiceResponse);
+    if inspectionLotData is error {
+        log:printError("Error while transforming inspection lot: " + inspectionLotData.message());
+        return;
+    }
+    io:print(inspectionLotData);
+
+    InspectionLot[] data = inspectionLotData._InspectionLotDetails;
+    inspectionLot:CreateA_InspectionLot inspectionLot = {
+        InspectionLot: inspectionLotData.InspectionLot,
+        InspectionLotType: inspectionLotData.InspectionLotType,
+        Material: data[0].Material,
+        Plant: data[0].Plant,
+        InspectionLotQuantity: data[0].Quantity.toString()
+    };
```

Comment:
Isnt this the transform code? Why are we transforming twice?

Reply:
Made the [improvement](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.qm/pull/24/commits/230ab97a4f3ea8e5edb4540319d92a08570c17ee)